math/wide_integer/uintwide_t.h

```diff
@@ -2116,17 +2159,17 @@
     }
 
     template<const bool RePhraseIsSigned = IsSigned,
-             typename std::enable_if<(RePhraseIsSigned == false)>::type const* = nullptr>
-    WIDE_INTEGER_CONSTEXPR std::int_fast8_t compare(const uintwide_t<Width2, LimbType, AllocatorType, RePhraseIsSigned>& other) const
+             typename std::enable_if<(!RePhraseIsSigned)>::type const* = nullptr>
+    WIDE_INTEGER_CONSTEXPR auto compare(const uintwide_t<Width2, LimbType, AllocatorType, RePhraseIsSigned>& other) const -> std::int_fast8_t
     {
       return compare_ranges(values.data(),
                             other.values.data(),
                             uintwide_t<Width2, LimbType, AllocatorType, RePhraseIsSigned>::number_of_limbs);
     }
 
     template<const bool RePhraseIsSigned = IsSigned,
-             typename std::enable_if<(RePhraseIsSigned == true)>::type const* = nullptr>
-    WIDE_INTEGER_CONSTEXPR std::int_fast8_t compare(const uintwide_t<Width2, LimbType, AllocatorType, RePhraseIsSigned>& other) const
```

Comment:
This is another situation where I doubt that `int` will lose you any performance or space. It really is designed to be the best all-round type. (Have you performed measurements or looked at the code gen?)